groovy-language-server/src/main/java/com/palantir/ls/util/Ranges.java

```diff
@@ -85,4 +85,32 @@ public static boolean contains(Range range, Position position) {
                 && POSITION_COMPARATOR.compare(range.getEnd(), position) >= 0;
     }
 
+    /**
+     * Returns the maximum position.
+     */
+    public static Position max(Position position1, Position position2) {
+        return POSITION_COMPARATOR.reversed().compare(position1, position2) < 0 ? position1 : position2;
```

Comment:
Maybe `POSITION_COMPARATOR.compare(position1, position2) >= 0` to be consistent with the function above? can ignore it, don't feel strongly about this